x-pack/legacy/plugins/lens/public/indexpattern_plugin/datapanel.tsx

```diff
@@ -163,18 +225,96 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
   }
 
   const allFields = indexPatterns[currentIndexPatternId].fields;
-  const filteredFields = allFields
+
+  const availableFieldTypes = uniq(allFields.map(({ type }) => type)).filter(
+    type => type in fieldTypeNames
+  );
+
+  const displayedFields = allFields
```

Comment:
I've updated this to use a map instead of doing the lookup each time and to simplify the double filtering, but I did not fully change the filtering logic like you're suggesting.